Modules/Scripted/DICOMLib/DICOMUtils.py

```diff
@@ -156,3 +156,180 @@ def __enter__(self):
+    for plugin in loadablesByPlugin:
+      for loadable in loadablesByPlugin[plugin]:
+        if loadable.selected:
+          if actualSelectedPlugins.has_key(plugin.loadType):
+            count = int(actualSelectedPlugins[plugin.loadType])
+            actualSelectedPlugins[plugin.loadType] = count+1
+          else:
+            actualSelectedPlugins[plugin.loadType] = 1
+    for pluginName in selectedPlugins.keys():
+      if not actualSelectedPlugins.has_key(pluginName):
+        logging.error("Expected DICOM plugin '%s' was not selected" % (pluginName))
+        success = False
+      elif actualSelectedPlugins[pluginName] != selectedPlugins[pluginName]:
+        logging.error("DICOM plugin '%s' was expected to be selected in %d loadables, but was selected in %d" % \
+          (pluginName, selectedPlugins[pluginName], actualSelectedPlugins[pluginName]))
+        success = False
+
+  # Count relevant node types in scene
+  actualLoadedNodes = {}
+  for nodeType in loadedNodes.keys():
+    nodeCollection = slicer.mrmlScene.GetNodesByClass(nodeType)
+    nodeCollection.UnRegister(None)
+    actualLoadedNodes[nodeType] = nodeCollection.GetNumberOfItems()
+
+  # Load selected data
+  dicomWidget.detailsPopup.loadCheckedLoadables()
+
+  for nodeType in loadedNodes.keys():
+    nodeCollection = slicer.mrmlScene.GetNodesByClass(nodeType)
+    nodeCollection.UnRegister(None)
+    numOfLoadedNodes = nodeCollection.GetNumberOfItems()-actualLoadedNodes[nodeType]
+    if numOfLoadedNodes != loadedNodes[nodeType]:
+      logging.error("Number of loaded %s nodes was %d, but %d was expected" % \
+        (nodeType, numOfLoadedNodes, loadedNodes[nodeType]) )
+      success = False
+
+  return success
+
+#------------------------------------------------------------------------------
+# Collect all series instance UIDs in a DICOM database (the Slicer one by default)
+# Useful to get list of just imported series UIDs, for example:
+# newSeriesUIDs = [x for x in seriesUIDsAfter if x not in seriesUIDsBefore]
+def allSeriesUIDsInDatabase(database=None):
+  if database is None:
+    database = slicer.dicomDatabase
+  dicomWidget = slicer.modules.dicom.widgetRepresentation().self()
+  allSeriesUIDs = []
+  for patient in database.patients():
+    studies = database.studiesForPatient(patient)
+    series = [database.seriesForStudy(study) for study in studies]
+    seriesUIDs = [uid for uidList in series for uid in uidList]
+    allSeriesUIDs.extend(seriesUIDs)
+  return allSeriesUIDs
+
+#------------------------------------------------------------------------------
+# Collect series instance UIDs belonging to a list of files
+def seriesUIDsForFiles(files):
+  seriesUIDs = set()
+  for file in files:
+    seriesUID = slicer.dicomDatabase.seriesForFile(file)
+    if seriesUID != '':
+      seriesUIDs.add(seriesUID)
+  return seriesUIDs
+
+#------------------------------------------------------------------------------
+class OnlineDICOMData:
```

Comment:
I'm wondering if we could find a more accurate name for this class.

I think usually there is a verb after `with` statement, such as:

    with open("x.txt") as f:
        ...

Also, OnlineDICOMData emphasizes that the data is online (vs offline), which is not that much important (we may enhance the class to be able to work from local files).

The name of the class does not give a hint about what it does (download, unzip, import, load, verify).

Maybe something like "LoadDICOMFilesToDatabase" would be a bit better (not emphasizing online/offline and would tell them ultimately the data is loaded).

Reply:
Sounds reasonable. But then the other context class in the utils called TemporaryDICOMDatabase should be renamed too, and it may be used outside Slicer core by now.

Reply:
These are the context managers in Slicer
* NodeModify
* TemporaryDICOMDatabase
* ExtensionProject